### PR TITLE
Use Docker to build the application to be more reproductive

### PR DESCRIPTION
**Pull request by sbrunner**

### CODE REVIEW

File: core/build.gradle [hunk 1]
```diff
@@ -238,7 +233,7 @@ war {
 }
 
 gretty {
-    // Configuration to serve mfp under localhost:8080/print when you run
+    // Configuration to serve mfp under print:8080/print when you run
```
Comment (danduk82) [inline core/build.gradle:236]: I think this is wrong

File: docker-compose.yaml [hunk 2]
```diff
@@ -0,0 +1,18 @@
+---
+
+version: '2.1'
+
+services:
+  geoserver:
+    image: camptocamp/geoserver
```
Comment (danduk82) [inline docker-compose.yaml:7]: maybe pin to a specific version

File: examples/README.md [hunk 3]
```diff
@@ -19,18 +19,17 @@ Test client
 
 The test server includes a client which can be used for testing. To start the server, run:
 
-    ./gradlew examples:farmRun
+    docker-compose up -d
 
-This will start a GeoServer at <http://localhost:8080/gs-web-app/> but also a MapFish Print with a simple
-client that can be accessed at <http://localhost:8080/print/>
+This will start a GeoServer at <http://geoserver:8080/> but also a MapFish Print with a simple
+client that can be accessed at <http://print:8080/>
```
Comment (danduk82) [inline examples/README.md:25]: specify that this works only inside a container within the same context

File: examples/src/test/java/org/mapfish/print/PrintApiTest.java [hunk 4]
```diff
@@ -515,7 +515,7 @@ private void assertRequiresAuth(String path) throws IOException, URISyntaxExcept
         ClientHttpRequest request = getRequest(path, HttpMethod.GET);
         HttpURLConnection urlConnection = (HttpURLConnection) request.getURI().toURL().openConnection();
         assertEquals(HttpStatus.FOUND.value(), urlConnection.getResponseCode());
-        assertEquals("https://localhost:8443/print/" + path, urlConnection.getHeaderField("Location"));
+        assertEquals("https://print:8443/" + path, urlConnection.getHeaderField("Location"));
```
Comment (danduk82) [inline examples/src/test/java/org/mapfish/print/PrintApiTest.java:518]: this port feels weird

Comment (sbrunner) [reply examples/src/test/java/org/mapfish/print/PrintApiTest.java:518]: Strange, but same as before...

**Comment by sbrunner** [comment]
@danduk82 updated :-)

**Pull request merged**